MetaMask/metamask-extension
# Dev olu hardware wallet feature flag

## Explanation
Put the hardware wallet behind a feature flag and disable/remove ledger, trezor and gridplus  connect buttons

* Fixes #16677
* Blocked By: [PR](https://github.com/MetaMask/KeyringController/pull/169)


## Screenshots/Screencaps

### Before

<img width="379" alt="Screenshot 2022-11-29 at 17 39 27" src="https://user-images.githubusercontent.com/44811/204543755-b1126066-02f6-48a4-8e09-9ace2cb735a8.png">


### After

<img width="385" alt="Screenshot 2022-11-29 at 17 44 53" src="https://user-images.githubusercontent.com/44811/204545076-badbc8ca-91b6-49eb-9093-b0550ac5164a.png">

## Manual Testing Steps

1. Start MetaMask in MV3 mode
2. Login
3. Click on Connect Hardware wallet
4. You should see the screen like in Before
5. Edit .metamaskrc
6. Add `HARDWARE_WALLETS_MV3=1`
7. Restart MetaMask in MV3 mode
8. Login
9. Click Connect Hardware wallet
10. You should see screen like in After
11. Restart MetaMask in MV2 mode
12. Login
13. Click Connect Hardware wallet
14. You should see screen like in Before

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
